core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1810,6 +1814,69 @@ private boolean isWindowTableFunctionRex(RexNode rex) {
     };
   }
 
+  /**
+   * A context that uses a select list.
+   */
+  final class SelectListContext extends BaseContext {
+    private final boolean aliasRef;
+    private final SqlNodeList selectList;
+
+    SelectListContext(
```

Comment:
I know you didn't write this class, but I can't quickly figure out what aliasRef means, if you know maybe you can add some JavaDoc.

Reply:
Yeah, no problem